portfolio/src/main/java/com/google/sps/data/FindSchedule.java

```diff
@@ -0,0 +1,116 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public final class FindSchedule {
+  private static final Comparator<CalendarEvent> sortAscending =
+      Comparator.comparingLong(CalendarEvent::getStartTimeLong);
+
+  private static final Comparator<Task> sortByDuration =
+      Comparator.comparing(Task::getDurationSeconds).thenComparing(Task::getName);
+
+  /**
+   * This method returns a Task Collection which the greedy algorithm
+   * determines should be scheduled for the person.
+   */
+  public Collection<Task> greedy(Collection<CalendarEvent> events, Collection<Task> tasks,
+      long workHoursStartTimeSeconds, long workHoursEndTimeSeconds) {
+    List<CalendarEvent> eventsList = new ArrayList<CalendarEvent>(events);
+    Task[] tasksArray = tasks.toArray(new Task[tasks.size()]);
+    Collections.sort(eventsList, sortAscending);
+    Arrays.sort(tasksArray, 0, tasks.size(), sortByDuration);
+    TimeRange[] availableTimes =
+        emptyTimeRanges(eventsList, workHoursStartTimeSeconds, workHoursEndTimeSeconds);
+
+    List<Task> scheduledTasks = new ArrayList<Task>();
+    // Index of TimeRange we are in
+    int i = 0;
+    // Index of Task we are trying to schedule
+    int j = 0;
+    // Integer indicating the start time (in seconds) we are currently trying to
+    // schedule events in
+    long scheduleTimeSeconds = 0;
+
+    // This will iterate through the time ranges and tasks and if one can be
+    // scheduled then it will be (and we move onto the next task) otherwise
+    // we move on to the next range (this is because the tasks are sorted by
+    // duration so if one task did not fit in the given range then we know no
+    // later ones will fit either). We create new Task objects for the result
+    // so data structures passed in are never changed.
+    while (i < availableTimes.length && j < tasksArray.length) {
+      TimeRange timeRange = availableTimes[i];
+      Task task = tasksArray[j];
+      // Either time is already past the start of the time range or we should
+      // update it (maybe this is our first iteration in the range).
+      scheduleTimeSeconds = Math.max(scheduleTimeSeconds, timeRange.start());
+      // The task can be scheduled in the current time range.
+      if (scheduleTimeSeconds + task.getDurationSeconds() <= timeRange.end()) {
+        Task scheduledTask;
+        if (task.getDescription().isPresent()) {
```

Comment:
It would be simpler to use ternary operator in the constructor.

ex:
Task scheduledTask = new Task(
    task.getName(),
    task.getDescription().isPresent() ? task....get() : null,
    ...);

Reply:
Refactored using a new class so this wasn't necessary.